centreon/src/Core/Media/Infrastructure/API/FindMedias/FindMediasController.php

```diff
@@ -19,22 +19,57 @@
  *
  */
 
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Core\Media\Infrastructure\API\FindMedias;
 
 use Centreon\Application\Controller\AbstractController;
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Media\Application\Exception\MediaException;
 use Core\Media\Application\UseCase\FindMedias\FindMedias;
 use Core\Media\Application\UseCase\FindMedias\FindMediasPresenterInterface;
 use Symfony\Component\HttpFoundation\Response;
 
 final class FindMediasController extends AbstractController
 {
+    use LoggerTrait;
+
+    /**
+     * @param ContactInterface $user
+     */
+    public function __construct(
+        readonly private ContactInterface $user,
+    ) {
+    }
+
     public function __invoke(FindMedias $useCase, FindMediasPresenterInterface $presenter): Response
     {
-        $this->denyAccessUnlessGrantedForApiConfiguration();
+        try {
+            $this->info('Access granted for media configuration.', ['user_id' => $this->user->getId()]);
+            if (
+                ! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_MEDIA_READ)
```

Comment:
Checking ACL rights is a business function and must therefore be included in the use case.